File: back/src/main/kotlin/com/jvmvstv_v/back/questionnaire/service/QuestionnaireService.kt
package com.jvmvstv_v.back.questionnaire.service

import com.jvmvstv_v.back.questionnaire.model.QuestionnaireKind
import com.jvmvstv_v.back.questionnaire.model.QuestionnaireVersion
import com.jvmvstv_v.back.questionnaire.model.SaveQuestionnaireResponseInput
import com.jvmvstv_v.back.questionnaire.model.SaveQuestionnaireTemplateInput
import com.jvmvstv_v.back.questionnaire.model.UserQuestionnaireResponse
import java.util.UUID

interface QuestionnaireService {
    fun activeQuestionnaire(kind: QuestionnaireKind): QuestionnaireVersion?
    fun questionnaireVersion(id: Int): QuestionnaireVersion?
    fun allVersions(kind: QuestionnaireKind): List<QuestionnaireVersion>
    fun myResponse(questionnaireVersionId: Int): UserQuestionnaireResponse?
    fun response(userId: UUID, questionnaireVersionId: Int): UserQuestionnaireResponse?
    fun saveResponse(input: SaveQuestionnaireResponseInput): UserQuestionnaireResponse
    fun submitResponse(id: UUID): UserQuestionnaireResponse
    fun saveTemplate(input: SaveQuestionnaireTemplateInput): QuestionnaireVersion
}

File: front/src/services/questionnaire/QuestionnaireService.ts
import type {
  QuestionnaireKind,
  QuestionnaireVersion,
  SaveQuestionnaireResponseInput,
  SaveQuestionnaireTemplateInput,
  UserQuestionnaireResponse,
} from './types.ts'

export interface QuestionnaireService {
  activeQuestionnaire(kind: QuestionnaireKind): Promise<QuestionnaireVersion | null>
  questionnaireVersion(id: string): Promise<QuestionnaireVersion | null>
  questionnaireVersions(kind: QuestionnaireKind): Promise<QuestionnaireVersion[]>
  myQuestionnaireResponse(questionnaireVersionId: string): Promise<UserQuestionnaireResponse | null>
  saveQuestionnaireResponse(input: SaveQuestionnaireResponseInput): Promise<UserQuestionnaireResponse>
  submitQuestionnaireResponse(id: string): Promise<UserQuestionnaireResponse>
  saveQuestionnaireTemplate(input: SaveQuestionnaireTemplateInput): Promise<QuestionnaireVersion>
}
